dkg-primitives/src/rounds/dkg.rs

```diff
@@ -0,0 +1,687 @@
+use curv::{arithmetic::Converter, elliptic::curves::Secp256k1, BigInt};
+use log::{error, info, trace, warn};
+
+use sc_keystore::LocalKeystore;
+use sp_core::sr25519;
+use sp_runtime::traits::AtLeast32BitUnsigned;
+use std::{collections::HashMap, path::PathBuf, sync::Arc};
+
+use super::{keygen::*, offline::*, sign::*};
+use std::mem::replace;
+
+use crate::types::*;
+use dkg_runtime_primitives::keccak_256;
+
+pub use gg_2020::{
+	party_i::*,
+	state_machine::{keygen::*, sign::*},
+};
+pub use multi_party_ecdsa::protocols::multi_party_ecdsa::{
+	gg_2020,
+	gg_2020::state_machine::{keygen as gg20_keygen, sign as gg20_sign, traits::RoundBlame},
+};
+
+/// DKG State tracker
+pub struct DKGState<C>
+where
+	C: AtLeast32BitUnsigned + Copy,
+{
+	pub accepted: bool,
+	pub is_epoch_over: bool,
```

Comment:
Would be better to use `epoch_is_over` over `is_epoch_over` here

In general:
> Do name Boolean properties with an affirmative phrase (CanSeek instead of CantSeek). Optionally, you can also prefix Boolean properties with Is, Can, or Has, but only where it adds value.